h2/src/main/org/h2/api/Interval.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2004-2018 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (http://h2database.com/html/license.html).
+ * Initial Developer: H2 Group
+ */
+package org.h2.api;
+
+import org.h2.util.DateTimeUtils;
+
+/**
+ * {@code INTERVAL} representation for result sets.
+ */
+public final class Interval {
+
+    private final IntervalQualifier qualifier;
+
+    private final boolean negative;
```

Comment:
what design drove the splitting of the negative flag separate from just letting leading be negative? Are we following some other model here?